vendor/sigs.k8s.io/kustomize/api/internal/git/repospec.go

```diff
@@ -192,76 +246,142 @@ func peelQuery(arg string) (string, string, time.Duration, bool) {
 		}
 	}
 
-	return parsed.Path, ref, duration, submodules
+	return ref, duration, submodules
 }
 
-func parseHostSpec(n string) (string, string) {
-	var host string
-	// Start accumulating the host part.
-	for _, p := range []string{
-		// Order matters here.
-		"git::", "gh:", "ssh://", "https://", "http://",
-		"git@", "github.com:", "github.com/"} {
```

Comment:
I just spot-checked this file, but I didn't see an equivalent to this ordering... is the new code behaviorally compatible with the old code?